uaclient/security.py

```diff
@@ -1379,17 +1422,23 @@ def upgrade_packages_and_attach(
             [
                 ["apt", "update", "&&"]
                 + ["apt", "install", "--only-upgrade", "-y"]
-                + sorted(upgrade_pkgs)
+                + sorted(pkg.split("=")[0] for pkg in upgrade_pkgs)
             ]
         )
     )
 
     if not dry_run:
-        apt.run_apt_update_command()
-        apt.run_apt_command(
-            cmd=["apt-get", "install", "--only-upgrade", "-y"] + upgrade_pkgs,
-            error_msg=messages.APT_INSTALL_FAILED.msg,
-            env={"DEBIAN_FRONTEND": "noninteractive"},
-        )
+        try:
+            apt.run_apt_update_command()
+            apt.run_apt_command(
+                cmd=["apt-get", "install", "--only-upgrade", "-y"]
+                + upgrade_pkgs,
+                env={"DEBIAN_FRONTEND": "noninteractive"},
+                capture=False,
+            )
+        except Exception as e:
```

Comment:
except Exception could be improved here, if in `run_apt*_command` we capture the subp error and raise a more specific one

Reply:
I think the idea here is to be more resilient to any possible exception. For example an error that is not caught by the `ProcessExecutionError`